scripts/eslint-rules/no-raw-global-usage.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+function reportUnexpectedGlobal(context, node) {
+  context.report({
+    node: node,
+    message:
+      'Unexpected use of {{global}}. Please import {{identifier}} from ' +
+      'shared/Globals to ensure safe access.',
+    data: {
+      identifier: node.property.name,
+      global: node.object.name,
+    },
+  });
+}
+
+module.exports = {
+  meta: {
+    schema: [],
+  },
+  create: function(context) {
+    if (context.getFilename().endsWith('/packages/shared/Globals.js')) {
+      return {};
+    }
+    let hasWindowImport = false;
+    return {
+      ImportDeclaration: function(node) {
+        const source = node.source;
+        if (source && source.value === 'shared/Globals') {
+          if (
+            node.specifiers &&
+            node.specifiers.find(binding => binding.local.name === 'window')
+          ) {
+            hasWindowImport = true;
+          }
+        }
+      },
+      MemberExpression: function(node) {
+        if (
+          node.object.type === 'Identifier' &&
+          ((node.object.name === 'window' && !hasWindowImport) ||
+            node.object.name === 'global') &&
+          node.property.type === 'Identifier'
+        ) {
+          reportUnexpectedGlobal(context, node);
```

Comment:
This should also catch any other globals (i.e. anything that is not imported) like `jest` or `navigator`. For that you can check the `scope`, like in `no-restricted-globals` [here](https://github.com/eslint/eslint/blob/master/lib/rules/no-restricted-globals.js#L101).

Reply:
I think flow already catches those types of errors. Do you think we need them caught by eslint too?

Reply:
Maybe I'm missing something. It should catch these:

<img width="808" alt="Screen Shot 2021-09-30 at 12 27 10 PM" src="https://user-images.githubusercontent.com/2440089/135494482-03f6f2d3-85b7-4437-a643-5a14697d2360.png">